sonic_db_config/db_config.go

```diff
@@ -1,135 +1,237 @@
+		if target == ns {
+			return target, true
+		}
+	}
+	return "", false
+}
+func GetDbList(ns string) map[string]interface{} {
+	if !sonic_db_init {
+		DbInit()
+	}
+	db_list, ok := sonic_db_config[ns]["DATABASES"].(map[string]interface{})
+	if !ok {
+		panic(fmt.Errorf("DATABASES' is not valid key in database_config.json file for namespace `%v` !", ns))
+	}
+	return db_list
+}
+
+func GetDbInst(db_name string, ns string) map[string]interface{} {
+	if !sonic_db_init {
+		DbInit()
+	}
+	db, ok := sonic_db_config[ns]["DATABASES"].(map[string]interface{})[db_name]
+	if !ok {
+		panic(fmt.Errorf("database name '%v' is not valid in database_config.json file for namespace `%v`!", db_name, ns))
+	}
+	inst_name, ok := db.(map[string]interface{})["instance"]
+	if !ok {
+		panic(fmt.Errorf("'instance' is not a valid field in database_config.json file for namespace `%v`!", ns))
+	}
+	inst, ok := sonic_db_config[ns]["INSTANCES"].(map[string]interface{})[inst_name.(string)]
+	if !ok {
+		panic(fmt.Errorf("instance name '%v' is not valid in database_config.json file for namespace `%v`!", inst_name, ns))
+	}
+	return inst.(map[string]interface{})
+}
+
+func GetDbSeparator(db_name string, ns string) string {
+	if !sonic_db_init {
+		DbInit()
+	}
+	db_list := GetDbList(ns)
+	separator, ok := db_list[db_name].(map[string]interface{})["separator"]
+	if !ok {
+		panic(fmt.Errorf("'separator' is not a valid field in database_config.json file!"))
+	}
+	return separator.(string)
+}
+
+func GetDbId(db_name string, ns string) int {
+	if !sonic_db_init {
+		DbInit()
+	}
+	db_list := GetDbList(ns)
+	id, ok := db_list[db_name].(map[string]interface{})["id"]
+	if !ok {
+		panic(fmt.Errorf("'id' is not a valid field in database_config.json file!"))
+	}
+	return int(id.(float64))
+}
+
+func GetDbSock(db_name string, ns string) string {
+	if !sonic_db_init {
+		DbInit()
+	}
+	inst := GetDbInst(db_name, ns)
+	unix_socket_path, ok := inst["unix_socket_path"]
+	if !ok {
+		panic(fmt.Errorf("'unix_socket_path' is not a valid field in database_config.json file!"))
+	}
+	return unix_socket_path.(string)
+}
+
+func GetDbHostName(db_name string, ns string) string {
+	if !sonic_db_init {
+		DbInit()
+	}
+	inst := GetDbInst(db_name, ns)
+	hostname, ok := inst["hostname"]
+	if !ok {
+		panic(fmt.Errorf("'hostname' is not a valid field in database_config.json file!"))
+	}
+	return hostname.(string)
+}
+
+func GetDbPort(db_name string, ns string) int {
+	if !sonic_db_init {
+		DbInit()
+	}
+	inst := GetDbInst(db_name, ns)
+	port, ok := inst["port"]
+	if !ok {
+		panic(fmt.Errorf("'port' is not a valid field in database_config.json file!"))
+	}
+	return int(port.(float64))
+}
+
+func GetDbTcpAddr(db_name string, ns string) string {
+	if !sonic_db_init {
+		DbInit()
+	}
+	hostname := GetDbHostName(db_name, ns)
+	port := GetDbPort(db_name, ns)
+	return hostname + ":" + strconv.Itoa(port)
+}
+
+func DbGetNamespaceAndConfigFile(ns_to_cfgfile_map map[string]string) {
```

Comment:
some functions use called `PrefixDb` notation and another part using `DbSuffix` notation.
For example `GetDbTcpAddr` and `DbGetNamespaceAndConfigFile`.
What is the accepted naming conversion in the project?

Reply:
changes are to make sure backward compatibility so  existing fuction do not break.
No particuar design here.